tests/Game.class.test.js

```diff
@@ -0,0 +1,965 @@
+      const game2048 = new Game([
+        [16, 8, 0, 0],
+        [0, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 0, 0, 16],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      game2048.getState().forEach(row => {
+        expect(row).toContain(16);
+      });
+    });
+
+    it('should merge multiple cells with the same value to the left', () => {
+      const game2048 = new Game([
+        [8, 8, 16, 16],
+        [16, 16, 8, 8],
+        [32, 32, 0, 0],
+        [16, 16, 64, 64],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).toBe(16);
+      expect(state[0][1]).toBe(32);
+      expect(state[1][0]).toBe(32);
+      expect(state[1][1]).toBe(16);
+      expect(state[2][0]).toBe(64);
+      expect(state[3][0]).toBe(32);
+      expect(state[3][1]).toBe(128);
+    });
+
+    it('should not merge multiple cells with the same value to the left multiple times during the move', () => {
+      const game2048 = new Game([
+        [8, 8, 0, 0],
+        [16, 16, 16, 16],
+        [32, 32, 64, 128],
+        [2, 2, 4, 8],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).toBe(16);
+      expect(state[1][0]).toBe(32);
+      expect(state[1][1]).toBe(32);
+      expect(state[2][0]).toBe(64);
+      expect(state[2][1]).toBe(64);
+      expect(state[2][2]).toBe(128);
+      expect(state[3][0]).toBe(4);
+      expect(state[3][1]).toBe(4);
+      expect(state[3][2]).toBe(8);
+    });
+
+    it('should not move cells to the left if the move does not change the board', () => {
+      const game2048 = new Game([
+        [64, 32, 16, 0],
+        [128, 64, 32, 0],
+        [256, 128, 64, 0],
+        [512, 256, 128, 0],
+      ]);
+
+      game2048.start();
+
+      const stateAfterStart = copyState(game2048.getState());
+
+      game2048.moveLeft();
+
+      expect(game2048.getState()).toEqual(stateAfterStart);
+    });
+  });
+
+  describe('moveRight', () => {
+    it('should not move the board if the game is not started', () => {
+      const initialState = [
+        [8, 0, 0, 0],
+        [0, 16, 0, 0],
+        [0, 0, 32, 0],
+        [0, 0, 0, 64],
+      ];
+
+      const game2048 = new Game(copyState(initialState));
+
+      game2048.moveRight();
+
+      expect(game2048.getState()).toEqual(initialState);
+    });
+
+    it('should move all non-zero cells to the right', () => {
+      const initialState = [
+        [8, 0, 0, 0],
+        [0, 16, 0, 0],
+        [0, 0, 32, 0],
+        [0, 0, 0, 64],
+      ];
+
+      const game2048 = new Game(initialState);
+
+      game2048.start();
+      game2048.moveRight();
+
+      const state = game2048.getState();
+
+      expect(state[0][3]).not.toBe(0);
+      expect(state[1][3]).not.toBe(0);
+      expect(state[2][3]).not.toBe(0);
+      expect(state[3][3]).not.toBe(0);
```

Comment:
The test case 'should move all non-zero cells to the right' has the same issue as the previous comment about moving cells to the left: the game is not started before the move.